osc-server/src/main/java/org/osc/core/broker/service/validator/DeleteDistributedApplianceRequestValidator.java

```diff
@@ -62,4 +73,26 @@ public DistributedAppliance validateAndLoad(BaseDeleteRequest request) throws Ex
 
         return da;
     }
+
+    private void validateDAReferenceToSfcAndSecurityGroup(EntityManager em, DistributedAppliance da) throws Exception {
+		for (VirtualSystem vs : da.getVirtualSystems()) {
+			if (!vs.getServiceFunctionChains().isEmpty()) {
+				// DE5296-->Look if these SFCs are binded to any of the SG of same DA
+				for (ServiceFunctionChain sfc : vs.getServiceFunctionChains()) {
+					List<SecurityGroup> sgList = SecurityGroupEntityMgr.listSecurityGroupsBySfcId(em, sfc.getId());
```

Comment:
Add this to a private method to improve readability: validateSfcReferenceToSecurityGroup. This combination of private methods will make the overall intended validation clearer